Net core global tools

johan-v-r opened the pull request:
Requires minimum [.NET Core SDK 2.1.300](https://www.microsoft.com/net/download/dotnet-core/2.1).

Changes include:
* `InheritDocLib` multi-target `netstandard2.0` and `net45`
* `InheritDoc` multi-target `netcoreapp2.1` and `net45`
* `InheritDoc` ability to pack [NuGet Global Tools](https://docs.microsoft.com/en-us/dotnet/core/tools/global-tools)

New packages can be built using [dotnet pack](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-pack?tabs=netcore2x#arguments).  

I'm not 100% sure how you currently build your NuGet package, however I was having some minor issues packing with the current `InheritDoc.nuspec` file's parameters. Perhaps it can be changed using the [NuGet metadata properties](https://docs.microsoft.com/en-us/dotnet/core/tools/csproj#nuget-metadata-properties), or doing something similar to [this comment](https://github.com/dotnet/cli/issues/2170#issuecomment-343414103) using `dotnet pack <path to .*proj file> /p:NuspecFile=<path to .nuspec file>`

johan-v-r commented on InheritDoc/InheritDoc/Program.cs:
```diff
@@ -1,76 +0,0 @@
-﻿/*
```
These files had to be deleted as they were conflicting with the new `csproj` style, which basically just includes all files in its directory. I guess they were probably more just for testing purposes or something..?

I've added this disclaimer into the `InheritDoc/Program.cs`.

firesharkstudios commented:
So, I...

- Committed the InheritDocVsix project (not sure why this had a .gitignore entry previously) which has an issue building
- Changed the properties on NLog.config so it gets copied to the output directories so InheritDoc has log output again
- Added the Powershell script used to build the NuGet packages

I'm really not up to speed on .NET Core -- not sure off hand how the NuGet packages need to be created for multiple targets.

firesharkstudios commented:
So, if we leave it to only target .NET Core 2.1, what will be the experience of someone installing InheritDoc NuGet package without .NET Core installed?

johan-v-r commented:
As this change will make it a _dotnet tool_, no project (regardless of .NET Framework or Core) would be able to install this as a dependency. They will receive the following error:  
> Package 'InheritDoc 1.2.3' has a package type 'DotnetTool' that is not supported by project ...

If this is a deal breaker, we could create another project in the same solution for something like `InheritDoc-Cli` which with a separate NuGet package for the tool.

To be honest, I see this application more as a development/build tool anyway (which is awesome), and personally wouldn't want it as a dependency on my application projects. I'd rather have it installed on my machine **once** and use on multiple projects as I need.

firesharkstudios commented:
I'd lean toward creating a new project called "InheritDocDotNetCore" that is targeted only at .NET Core.  Calling it "InheritDoc-Cli" will create confusion as "InheritDoc" is a command line tool at would look like you have to install InheritDoc-Cli to get a command line version of InheritDoc.

It might be in the near future that we figure out how to create a single NuGet package and eliminate the need to have a separate package.

Can you create a separate InheritDocDotNetCore project that targets only .NET Core and we'll revert InheritDoc to only target .NET Framework as it did before?

johan-v-r commented:
Okay I've updated the `.csproj` file with 2 `PropertyGroup` sections that has a condition for building with `GlobalTool`. By default (when emitted or `GlobalTool=false`) it only builds to .NET Framework 4.5 and includes the contents from `bin\$(Configuration)\$(TargetFramework)\` (new `ItemGroup` section) to the _tools_ folder in the NuGet package.

The `<PropertyGroup Condition="'$(GlobalTool)' == true">` section will build and package the .NET Core 2.1 global tool with a `<PackageId>InheritDoc-Tool</PackageId>`. So users will be able to install using `dotnet tool install -g inheritdoc-tool`, but still use it like `inheritdoc --version`. I've also added a description to indicate this, but you can change how you want.

![nuget](https://user-images.githubusercontent.com/13960403/44528088-f8b47680-a6e8-11e8-94c7-1829a3e016b4.png)

I decided to go with these sections in the same `csproj` rather than a new project and having to copy/extract your existing code.

firesharkstudios commented:
I like it, nice job.

Committed minor changes myself.  Will test a bit more, merge, and release if all goes well.
